src/LanguageServer/Protocol/Handler/TextDocumentContent/TextDocumentContentHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.TextDocumentContent;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(TextDocumentContentHandler)), Shared]
+[Method(Methods.WorkspaceTextDocumentContentName)]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class TextDocumentContentHandler() : ILspServiceDocumentRequestHandler<TextDocumentContentParams, TextDocumentContentResult>
```

Comment:
I wonder if we're also able to use this method for source generated documents too.